inspirehep/modules/search/query_factory.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""INSPIRE Query class to wrap the Q object from elasticsearch-dsl."""
+
+from __future__ import absolute_import, print_function
+
+import json
+import pypeg2
+
+from flask import current_app
+
+from elasticsearch_dsl import Q
+
+from werkzeug.utils import import_string
+
+from invenio_query_parser.ast import MalformedQuery
+
+from .walkers.elasticsearch import ElasticSearchDSL
+from .walkers.elasticsearch_no_keywords import ElasticSearchNoKeywordsDSL
+from .walkers.elasticsearch_no_keywords import QueryHasKeywords
+
+
+def inspire_query_factory():
+    """Create a parser returning Elastic Search DSL query instance."""
+
+    def invenio_query(pattern):
+        parser = import_string(current_app.config['SEARCH_QUERY_PARSER'])
```

Comment:
This would be called every time the function is called. Is there some performance penalty here using `import_string` every time? Same below regarding `walkers`.